spring-cloud-sleuth-brave/src/main/java/org/springframework/cloud/sleuth/brave/instrument/web/BraveHttpConfiguration.java

```diff
@@ -0,0 +1,291 @@
+		}
+
+		private brave.http.HttpResponseParser toBraveHttpResponseParser(BeanFactory beanFactory, String beanName) {
+			Object bean = beanFactory.getBean(beanName);
+			brave.http.HttpResponseParser parser = bean instanceof brave.http.HttpResponseParser
+					? (brave.http.HttpResponseParser) bean : bean instanceof HttpResponseParser
+							? BraveHttpResponseParser.toBrave((HttpResponseParser) bean) : null;
+			return returnOrThrow(bean, parser, beanName, brave.http.HttpResponseParser.class, HttpResponseParser.class);
+		}
+
+		private SamplerFunction<brave.http.HttpRequest> toBraveSampler(BeanFactory beanFactory, String beanName) {
+			Object bean = beanFactory.getBean(beanName);
+			SamplerFunction<brave.http.HttpRequest> braveSampler = bean instanceof SamplerFunction
+					? (SamplerFunction<brave.http.HttpRequest>) bean
+					: bean instanceof org.springframework.cloud.sleuth.api.SamplerFunction
+							? BraveSamplerFunction.toHttpBrave(
+									(org.springframework.cloud.sleuth.api.SamplerFunction<org.springframework.cloud.sleuth.api.http.HttpRequest>) bean)
+							: null;
+			return returnOrThrow(bean, braveSampler, beanName, SamplerFunction.class,
+					org.springframework.cloud.sleuth.api.SamplerFunction.class);
+		}
+
+		@NotNull
+		private <T> T returnOrThrow(Object bean, T convertedBean, String name, Class brave, Class sleuth) {
+			if (convertedBean == null) {
+				throw new IllegalStateException(
+						"Bean with name [" + name + "] is of type [" + bean.getClass() + "] and only ["
+								+ brave.getCanonicalName() + "] and [" + sleuth.getCanonicalName() + "] are supported");
+			}
+			return convertedBean;
+		}
+
+		private SamplerFunction<brave.http.HttpRequest> combineUserProvidedSamplerWithSkipPatternSampler(
+				@Nullable SamplerFunction<brave.http.HttpRequest> serverSampler,
+				@Nullable SkipPatternProvider provider) {
+			SamplerFunction<brave.http.HttpRequest> skipPatternSampler = provider != null
+					? new SkipPatternHttpServerSampler(provider) : null;
+			if (serverSampler == null && skipPatternSampler == null) {
+				return SamplerFunctions.deferDecision();
+			}
+			else if (serverSampler == null) {
+				return skipPatternSampler;
+			}
+			else if (skipPatternSampler == null) {
+				return serverSampler;
+			}
+			return new CompositeHttpSampler(skipPatternSampler, serverSampler);
+		}
+
+		@Bean
+		@ConditionalOnMissingBean(name = HttpClientSampler.NAME)
+		SamplerFunction<brave.http.HttpRequest> sleuthHttpClientSampler(SleuthWebProperties sleuthWebProperties) {
+			String skipPattern = sleuthWebProperties.getClient().getSkipPattern();
+			if (skipPattern == null) {
+				return SamplerFunctions.deferDecision();
+			}
+
+			return new SkipPatternHttpClientSampler(Pattern.compile(skipPattern));
+		}
+
+	}
+
+}
+
+/**
+ * Composite Http Sampler.
+ *
+ * @author Adrian Cole
+ */
+final class CompositeHttpSampler implements SamplerFunction<brave.http.HttpRequest> {
```

Comment:
Did you mean to nest those in `BraveHttpConfiguration`? Because it doesn't at the moment and having 4 top-level classes within the same `.java` file seems a bit strange to me.